### PR TITLE
Convert Register a Birth flow tests

### PR DESCRIPTION
**Pull request by leenagupte**
Trello: https://trello.com/c/oOTnSdsG

# What's changed and why?

Converts the Register a Birth flow tests to use the node based testing approach introduced in [ADR-3][1].

Also removes the existing flow integration tests as they are no longer needed.

The outcomes contain a lot of logic so they have been tested in separate commits.

Also tidies up the outcomes by removing unnecessary conditionals as well as conditionals and content that can never be reached.

[1]: https://github.com/alphagov/smart-answers/blob/main/docs/arch/003-testing-flows.md

⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

### CODE REVIEW

File: test/flows/register_a_birth_flow_test.rb [hunk 1]
```diff
@@ -0,0 +1,542 @@
...
+        assert_rendered_outcome text: "It’s illegal in Saudi Arabia to have a child outside of marriage"
+      end
+
+      should "render United Arab Emirates guidance if child born in United Arab Emirates" do
+        add_responses country_of_birth?: "united-arab-emirates"
+        assert_rendered_outcome text: "It’s illegal in the United Arab Emirates to have a child outside of marriage"
+      end
+    end
+
+    should "render other country guidance if birth not registered in country of birth" do
```
Comment (Rosa-Fox) [inline test/flows/register_a_birth_flow_test.rb:536]: I looked back to marriage abroad and Kevin used this `random_country` [method](https://github.com/alphagov/smart-answers/blob/ba95009797fdf422936b81d8e8f9a478cc8523f4/test/flows/marriage_abroad_flow_test.rb#L73) or looped [through each country](https://github.com/alphagov/smart-answers/blob/ba95009797fdf422936b81d8e8f9a478cc8523f4/test/flows/marriage_abroad_flow_test.rb#L183) so wondering if we should do one of those here, to catch the unlikely event it disappears for one of the countries in question?

File: app/flows/register_a_birth_flow/outcomes/no_birth_certificate_result.erb [hunk 2]
```diff
@@ -43,28 +43,7 @@
...
-    <% when 'pakistan' %>
-      You can’t get a local birth certificate if you weren’t married when you had your child.
-    <% when 'qatar' %>
-      You can’t get a local birth certificate if you weren’t married when you had your child.
-    <% when 'saudi-arabia' %>
-      You can’t get a local birth certificate if you weren’t married when you had your child.
-    <% when 'united-arab-emirates' %>
-      You can’t get a local birth certificate if you weren’t married when you had your child.
-    <% end %>
+    You can’t get a local birth certificate if you weren’t married when you had your child.
```
Comment (Rosa-Fox) [inline app/flows/register_a_birth_flow/outcomes/no_birth_certificate_result.erb:46]: Some good drying!

File: app/flows/register_a_birth_flow/outcomes/oru_result.erb [hunk 3]
```diff
@@ -246,21 +240,7 @@ Depending on the document, you'll be asked to provide an original version or a p
...
-
-    - more information
-    - to verify your documents
-    - the child or parents to take a DNA test
-
-    If this happens, it will take longer for the birth to be registered.
-
-    ^You must pay all costs relating to DNA testing.^
-  <% elsif calculator.custom_waiting_time %>
+  <% if calculator.custom_waiting_time %>
```
Comment (Rosa-Fox) [inline app/flows/register_a_birth_flow/outcomes/oru_result.erb:243]: 👍

File: app/flows/register_a_birth_flow/outcomes/oru_result.erb [hunk 4]
```diff
@@ -288,8 +268,6 @@ Depending on the document, you'll be asked to provide an original version or a p
         Your documents will be returned to the British High Commission in Lagos after the birth has been registered.
 
         You’ll also be sent copies of the registration certificate if you’ve paid for them. The High Commission will contact you when your documents and certificates are ready for collection.
-      <% when 'north-korea' %>
```
Comment (Rosa-Fox) [inline app/flows/register_a_birth_flow/outcomes/oru_result.erb:291]: 👍

**Review by Rosa-Fox — COMMENTED** [review]
Nice work @leenagupte! So many outcomes 😱! I left a couple comments, let me know what you think :)

File: test/flows/register_a_birth_flow_test.rb [hunk 5]
```diff
@@ -0,0 +1,542 @@
...
+
+    should "render court order guidance if birth registered in Indonesia, father is british and parents not married or in civil partnership" do
+      add_responses who_has_british_nationality?: "father",
+                    married_couple_or_civil_partnership?: "no",
+                    where_are_you_now?: "another_country",
+                    which_country?: "indonesia"
+      assert_rendered_outcome text: "You’ll need to get a court order"
+    end
+
+    should "render Libya guidance if birth registered in Libya" do
```
[the remark this answers is not in the document]
Comment (leenagupte) [reply test/flows/register_a_birth_flow_test.rb:219]: Good spot about [line](https://github.com/alphagov/smart-answers/blob/main/app/flows/register_a_birth_flow/outcomes/oru_result.erb#L23) I'll add a test for it and fix up.

I didn't add a test for [line](https://github.com/alphagov/smart-answers/pull/5581/commits/1a6e410a5c654969016ead1bf06bf7f3338a62f1) as there's nothing unique about it to test. All of the content in this section also appears for Libya, but does have a few extra words that can be tested.

File: test/flows/register_a_birth_flow_test.rb [hunk 6]
```diff
@@ -0,0 +1,542 @@
...
+
+    should "render Philippines guidance if child born in Philippines" do
+      add_responses country_of_birth?: "philippines"
+      assert_rendered_outcome text: "photos of the child as a baby, in which the child can clearly be identified with its parents in the Philippines"
+    end
+
+    should "render Uganda guidance if child born in Uganda" do
+      add_responses country_of_birth?: "uganda"
+      assert_rendered_outcome text: "photos of the child as a baby, in which the child can clearly be identified with its parents in Uganda"
+    end
```
[the remark this answers is not in the document]
Comment (leenagupte) [reply test/flows/register_a_birth_flow_test.rb:385]: I didn't add another test for Sierra Leone as there's already a test for it further up. I can add another test for it though, so we also test for non-documents guidance.

File: test/flows/register_a_birth_flow_test.rb (hunk 1 above)
Comment (leenagupte) [reply test/flows/register_a_birth_flow_test.rb:536]: I'll remove this commit and leave the loop in.

**Comment by leenagupte** [comment]
@Rosa-Fox Thanks for your review. I've addressed your comments and fixed up.

File: test/flows/register_a_birth_flow_test.rb (hunk 5 above)
[the remark this answers is not in the document]
Comment (Rosa-Fox) [reply test/flows/register_a_birth_flow_test.rb:219]: ah yeah I see what you mean about the text not having anything unique. Thanks for adding the test for the other line.

File: test/flows/register_a_birth_flow_test.rb (hunk 6 above)
[the remark this answers is not in the document]
Comment (Rosa-Fox) [reply test/flows/register_a_birth_flow_test.rb:394]: Thanks!

File: test/flows/register_a_birth_flow_test.rb (hunk 1 above)
Comment (Rosa-Fox) [reply test/flows/register_a_birth_flow_test.rb]: 👍

**Review by Rosa-Fox — APPROVED** [review]
Looks good!

**Pull request merged**